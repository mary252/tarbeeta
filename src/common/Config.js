// export const baseURL = "https://api.tarbeeta.com/v1/";
// export const v2_url = "https://api.tarbeeta.com/v2/";
export const v2_url = "https://staging.tarbeeta.com/v2/";
export const baseURL = "https://staging.api.tarbeeta.com/v1/";
export const socketUrl = "https://socket.tarbeeta.com:8878";
// export const baseURL = "http://192.168.1.15:3000/v1/";
// export const baseURL = "http://197.167.99.223:3126/v1/";
 //export const baseURL = "http://msi-laptop:3000/v1/";
export const assetUrl = "https://static.tarbeeta.com/images/";
export const baseUrlStatic = "https://www.tarbeeta.com";

//export const google_capcha="6Lf34qkUAAAAAKBh0Qs_9OZrBeZhikwqF6L7G62p"
// export const google_capcha="6LfE5akUAAAAAHC3KCKqJh1NGup58xJwQbjZxYgg"
export const google_capcha = "6LfXz6oUAAAAABbSRcTAPsK971UiPQZ6CpSSPWr4";
export const EN = 1;
export const AR = 2;
export const EG = 1;
export const DEFAULT_LOCALE = "ar";

// Facebook
export const FACEBOOK_APP_ID = "520130485093535";
export const FACEBOOK_PAGE_ID = "1996928467010956";
export const FACEBOOK_CHAT_POPUP_THEME_COLOR = "#5a9ae5";

export const PRODUCT_SLIDER_LIMIT = 18;

export const ORDER_STATUS_FAILED = 1;
export const ORDER_STATUS_PLACED = 2;
export const ORDER_STATUS_PAID = 3;
export const ORDER_STATUS_PENDING_PAYMENT = 4;
export const ORDER_STATUS_PENDING_SHIPPING = 5;
export const ORDER_STATUS_ON_THE_WAY = 6;
export const ORDER_STATUS_DELIVERED = 7;
export const ORDER_STATUS_CANCELLED = 8;
export const ORDER_STATUS_PENDING = 9;

export const APPROVED_PRODUCT = 4;

export const CASH_ON_DELEVERY = 4;
export const PAY_WALLET = 3;
export const MASTER_CARD = 1;
export const KIOSK = 2;

export const SORT_MOST_RECENT = 1;
export const SORT_OLD_TO_NEW = 2;
export const SORT_HIGHEST_PRICE = 3;
export const SORT_LOWEST_PRICE = 4;
export const SORT_RATING = 5;
export const CARD_PAYMENT_IFRAME = `https://accept.paymobsolutions.com/api/acceptance/iframes/`;
export const EN_FRAME_ID = 7340;
export const AR_FRAME_ID = 7459;
export const pending_product = 1;
