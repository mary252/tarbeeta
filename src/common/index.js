export * from "./Images";
export * from "./Config";
