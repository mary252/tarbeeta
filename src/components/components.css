.notification-raw-continer {
  padding: 13px;
  display: flex;
  align-items: center;
  flex-direction: row;
  border-radius: 5px;
  background-color: #ffffff;
  margin-bottom: 8px !important;
  position: relative;
}

.notification-raw-avatar {
  width: 29.2px;
  height: 29.2px;
  margin-right: 12px !important;
}

.notification-raw-time {
}

.noti-popup-text {
  font-size: 12px;
  color: #444444;
}
.noti-popup-time {
  font-size: 12px;
  line-height: 1.17;
  letter-spacing: normal;
  color: #bebebe;
}
.animate-loader {
  width: 300px !important;
  height: 225px !important;
}

.txt-centered {
  text-align: center;
  vertical-align: middle;
  margin-bottom: 5px;
}

.address-popup-div {
  border-radius: 5px;
  background-color: #ffffff;
  padding: 20px;
  height: fit-content;
  height: intrinisic;
  height: -moz-max-content;
}

.address-popup-header {
  font-size: 16px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: center;
  color: #444444;
  padding-top: 5px;
  margin-bottom: 25px;
}

@media screen and (max-width: 550px) {
  .address-popup-div {
    height: 100%;
  }
}
.table-border {
  border-top: 2px solid #e4eaee;
  border-bottom: 2px solid #e4eaee;
  padding: 10px 0;
}
.table-border {
  border-top: none;
  border-bottom: none;
  padding: 0;
}

.table-border > div:first-child {
  display: none;
}
.address-notes-mobile {
  text-align: center;
}
.address-notes-mobile p {
  word-break: break-all;
}

.table-head {
  padding: 20px;
}

.table-head h4 {
  margin-right: 20px;
  color: #444444;
  white-space: nowrap;
}

.table-row {
  border-bottom: none;
  padding: 11px 8px 16px 0;
  margin-bottom: 19px !important;
  border-bottom: 2px solid #e4eaee;
}
.nthse-odd {
  padding: 8px;
}
.table-row .nthse-odd:nth-of-type(even) {
  background: #edeced;
}


.table-row .columns .column h4,
.table-row .columns .column p {
  color: #555;
  font-size: 14px;
}

.address-num {
  font-weight: 500 !important;
  color: #555555;
  padding-left: 27px;
  padding-right: 27px;
}

/* .white-box-table {
    background: #fff;
    border-radius: 5px;
    padding: 15px 0;
} */

/* 1087px */

@media screen and (max-width: 1088px) {
  .shop-avatar {
    width: 80px;
    height: 80px;
  }

  .shop-cover,
  .drop-zone > div {
    height: 127px;
  }

  .drop-zone-content p {
    text-align: center;
  }

  .saved-address-text {
    margin-right: 12px;
    color: #444444;
  }

  .address-num {
    padding-left: 0;
    font-weight: 500 !important;
  }

  .table-row .columns .column:first-child div {
    border-top: 1px solid #e5e5e5;
    border-bottom: 1px solid #e5e5e5;
    height: 34px;
    margin-bottom: 10px;
    align-items: center;
    justify-content: center;
  }

  .table-row .columns .column {
    text-align: center;
  }

  .table-row .columns .column span {
    display: block;
    margin-bottom: 10px;
    color: #999;
    font-size: 12px;
    text-align: center;
  }

  .table-row .columns .column h4 {
    color: #555;
    font-size: 12px;
    text-align: center;
  }

  .table-head h4 {
    font-size: 14px;
  }

  .delete-btn-mobile {
    text-align: right;
  }

  .edit-btn-mobile {
    text-align: left;
  }

  .table-row {
    border-bottom: 0px solid #e5e5e5;
    padding: 0;
    margin-bottom: 0 !important;
  }

  .edit-address-section > div:nth-last-child(2) {
    margin-bottom: 19px !important;
    border-bottom: 1px solid #e5e5e5;
  }

  .table-row .columns .column p {
    text-align: center;
  }
}
