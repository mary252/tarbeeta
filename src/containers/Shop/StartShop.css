.checkbox-text {
  font-family: Rubik;
  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
}

.shop-avatar {
  margin-bottom: 7px;
  width: 140px;
  height: 140px;
  background: #fff;
  border-radius: 50%;
  background-size: contain !important;
  background-position: center !important;
  border: 1px solid #d8d8d8;
  background-repeat: no-repeat !important;
}

.settings-avatar {
  position: relative;
}

.settings-avatar button {
  right: 40px !important;
  top: 5px !important;
}

.shop-cover {
  margin-bottom: 22px;
  height: 300px;
  background-size: cover !important;
  background-position: center !important;
  background-repeat: no-repeat;
  border-radius: 5px;
  border: solid 1px #d8d8d8;
  max-width: 633px;
  background-repeat: no-repeat !important;
  position: relative;
}

.drop-zone {
  height: 100%;
  cursor: pointer;
  background-size: contain !important;
  position: relative;
}

.drop-zone div,
.drop-zone > div > input {
  outline: none !important;
}

.drop-zone > div {
  height: 300px;
}

.drop-zone-content {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  flex-direction: column;
}

.checkbox-text {
  width: 483px;
  /* height: 17px; */
  font-size: 14px;
  color: #565656;
}


/* .white-box-table {
    background: #fff;
    border-radius: 5px;
    padding: 15px 0;
} */

/* 1087px */

@media screen and (max-width: 1088px) {
  .shop-avatar {
    width: 80px;
    height: 80px;
  }

  .shop-cover,
  .drop-zone > div {
    height: 127px;
  }

  .drop-zone-content p {
    text-align: center;
  }

  .saved-address-text {
    margin-right: 12px;
    color: #444444;
  }

  .delete-btn-mobile {
    text-align: right;
  }

  .edit-btn-mobile {
    text-align: left;
  }

}
