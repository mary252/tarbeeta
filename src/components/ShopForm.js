import React, { Component } from "react";
import { isLoggedIn } from "../ultils/auth";
import { Redirect } from "react-router-dom";
import {
  TextInput,
  TextBox,
  Checkbox,
  TButton,
  Notification,
  VerificationPopup,
  AddressTable
} from "../components";

import Dropzone from "react-dropzone";
import { getShopId } from "../ultils/main";
import MessengerCustomerChat from 'react-messenger-customer-chat';
import { FACEBOOK_CHAT_POPUP_THEME_COLOR, FACEBOOK_APP_ID, FACEBOOK_PAGE_ID } from "../common";
import CustomCloseBtn from "../containers/Shop/CustomCloseBtn";

export class ShopForm extends Component {
  state = {
    loggedin: false,
    is_busy: false,
    isError: false,
    redirect: false,
    agreeTerms: true
  };

  copyLink = () => {
    if (this.state.username !== "") {
      navigator.clipboard.writeText(
        `https://www.tarbeeta.com/${this.props.lang}/${this.props.username}`
      );
      this.props.init_notification({
        type: "success",
        title: this.props.locale.success,
        message: this.props.locale.copied_msg
      });
    } else {
      navigator.clipboard.writeText(``);
    }
  };


  toggle_busy = () => this.setState({ is_busy: !this.state.is_busy });
  componentWillMount() {
    if (getShopId())
      this.props.history.push(`/${this.props.lang}/shop/edit`);
    if (isLoggedIn()) {
      this.setState({
        loggedin: true
      });
    }
  }

  VerificationPopup = null;
  openPopUp = () => this.VerificationPopup.toggle();


  render() {
    if (!this.state.loggedin) {
      let path = this..location.pathname;
      let redirect = path.substring("4");
      return <Redirect to={`/${this.props.lang}/login?src=${redirect}`} />;
    }
    if (this.state.redirect) {
      return <Redirect to={`/${this.props.lang}/mobile`} />;
    }
    const {
      state: { agreeTerms },
      props: {
            startShop,
            editShop,
            mobile_verified,
            verified,
            startYourShop,
            onChange,
            validation,
            nameError,
            usernameError,
            onDrop,
            changeAvatar,
            avatarPreview,
            avatar,
            cover,
            coverPreview,
            is_busy,
            message,
            isError,
            locale,
            username,
            bio,
            name,
            address,
            openPopUp,
            updateAddress,
            editYourShop,
            removeCover,
            onClose,
            delAddress,
            mobile,
            mobileError
        }
    } = this;

    return (
      <div>
        <Notification message={message} isError={isError} onClose={onClose} />

        <div>
            <MessengerCustomerChat
              pageId={FACEBOOK_PAGE_ID}
              appId={FACEBOOK_APP_ID}
              greeting_dialog_display="hide"
              themeColor={FACEBOOK_CHAT_POPUP_THEME_COLOR}
              language={this.props.lang == "en" ? "en_US" : "ar_AR"}
              greetingDialogDelay={0}
              minimized={true}
              shouldShowDialog={false}
              loggedInGreeting={this.props.locale.facebook_chat_greeting_message}
              loggedOutGreeting={this.props.locale.facebook_chat_greeting_message}
            />
        </div>

        <VerificationPopup
          ref={a => (this.VerificationPopup = a)}
          lang={this.props.lang}
          translation={this.props.locale}
          code={1}
        />

        <div className="section">
          <div className="container">
            <h3 className="page-title mar-bot-40">
              {startShop
                ? locale.shop_containers.start_shop_page_title
                : locale.shop_containers.edit_shop_page_title}
            </h3>
            <section className="white-box">
              <div className="columns">
                <div className="column is-5">
                  <div className="columns is-mobile is-flex is-vcentered">
                    <div className="column is-4-desktop is-4-mobile is-4-tablet">
                      <div
                        className="shop-avatar is-flex aic jcc"
                        style={{
                          

                          // background: startShop
                          //   ? avatar !== ""
                          //     ? `url(${avatarPreview}) no-repeat`
                          //     : null
                          //   : avatarPreview == ""
                          //   ? `url(${avatar}) no-repeat`
                          //   : `url(${avatarPreview})`

                          background: 
                             avatar !== ""
                              ? `url(${avatarPreview}) no-repeat`
                              : null

                              
                          // backgroundSize: avatar !== "" ? "contain" : "unset"
                        }}
                      >
                        {avatar == "" && (
                          <img
                            alt="icon"
                            src={require("../assets/images/image-file-icon.svg")}
                          />
                        )}

                        {avatar == null && (
                          <img
                            alt="icon"
                            src={require("../assets/images/image-file-icon.svg")}
                          />
                        )}
                      </div>
                    </div>
                    <div className="column is-8-desktop is-7-mobile is-8-tablet">
                      <div className="columns is-mobile">
                        <div className="column is-7-desktop is-12-mobile">
                          <TButton
                            text={locale.choose_picture_placeholder}
                            className="blue outline"
                            id={"avatar-button"}
                            name="avatar"
                            onPress={() => this.refs.changeAvatar.click()}
                            ariaLabel="Upload shop Photo"
                          />
                          <input
                            onChange={e => changeAvatar(e)}
                            type="file"
                            ref="changeAvatar"
                            className="is-hidden"
                          />
                        </div>
                      </div>
                      <p>{locale.shop_containers.shop_avatar_helper_p}</p>
                    </div>
                  </div>
                  <div className="columns" style={{ marginBottom: 0 }}>
                    <div className="column is-half">
                      <TextInput
                        placeholder={locale.shop_containers.shop_name_input_placeholder}
                        id={"name-input"}
                        onChange={e => onChange(e)}
                        name="name"
                        type="text"
                        errorMessage={nameError}
                        onBlur={() =>
                          validation(["name"])
                            .then()
                            .catch(e => {})
                        }
                        value={name}
                      />
                    </div>
                    <div className="column is-half">
                      <TextInput
                        placeholder={locale.shop_containers.shop_phone_placeholder}
                        id={"mobile"}
                        onChange={e => onChange(e)}
                        name="mobile"
                        type="text"
                        errorMessage={mobileError}
                        onBlur={() =>
                          validation(["mobile"])
                            .then()
                            .catch(e => {})
                        }
                        value={mobile}
                      />
                    </div>
                  </div>
                  <TextBox
                    placeholder={locale.shop_containers.shop_bio_placeholder}
                    rows="5"
                    id={"bio-input"}
                    custclass={`mar-bot-15`}
                    onChange={e => onChange(e)}
                    name="bio"
                    maxLength={160}
                    value={bio}
                  />
                  {/* <div className="columns" style={{ marginBottom: 0 }}>
                    <div className="column is-6">
                      <TextInput
                        placeholder={locale.shop_phone_placeholder}
                        id={"shop-phone"}
                        onChange={e => onChange(e)}
                        name="shop-phone"
                        type="text"
                        errorMessage={nameError}
                        onBlur={() =>
                          validation(["shop-phone"])
                            .then()
                            .catch(e => {})
                        }
                        value={shop_phone}
                      />
                    </div>
                  </div> */}
                  <div
                    className={`columns is-flex is-vcentered is-mobile is-multiline ${
                      this.props.lang == "ar" ? "rtl" : ""
                    }`}
                  >
                    <div className="column is-5-desktop is-12-mobile">
                      <p className={this.props.lang == "ar" ? "ltr is-flex jce" : ""}>https://www.tarbeeta.com/</p>
                    </div>
                    <div className="column is-4-desktop is-8-mobile">
                      <TextInput
                        placeholder={locale.shop_containers.shop_url_username_placeholder}
                        onChange={e => onChange(e)}
                        name="username"
                        value={username}
                        id="username"
                        type="text"
                        ref={textarea => (this.textArea = textarea)}
                        errorMessage={usernameError}
                        onBlur={() =>
                          validation(["username"])
                            .then()
                            .catch(e => {})
                        }
                      />
                    </div>
                    <div className="column is-3-desktop is-4-mobile">
                      <a
                        onClick={this.copyLink}
                        className="blue med"
                        href="javascript:void(0)"
                      >
                        {locale.shop_containers.shop_copy_link}
                      </a>
                    </div>
                  </div>
                </div>
                <div className="column is-1 is-hidden-mobile" />
                <div className="column is-6">
                  <div
                    className="shop-cover"
                    style={{
                      


                        // background: startShop
                        // ? cover !== ""
                        //   ? `url(${coverPreview}) no-repeat`
                        //   : null
                        // : coverPreview === ""
                        // ? `url(${cover}) no-repeat`
                        // : `url(${coverPreview})`,


                        background: 
                         cover !== ""
                          ? `url(${coverPreview}) no-repeat`
                          : null


                      
                        
                      // border: cover != "" ? "none" : null
                    }}
                  >
                    <Dropzone
                      id="dropzone"
                      name="dropzone"
                      getFilesFromEvent={e => onDrop(e)}
                    >
                      {({ getRootProps, getInputProps }) => (
                        <section className="drop-zone">
                          <div {...getRootProps()}>
                            <input {...getInputProps()} />
                            {cover == "" && (
                              <div className="drop-zone-content">
                                <div className="is-flex aic mar-bot-10">
                                  <img
                                    alt="icon"
                                    className="mar-right-20"
                                    src={require("../assets/images/image-file-icon.svg")}
                                  />
                                  <img
                                    alt="icon"
                                    src={require("../assets/images/drag-file-icon.svg")}
                                  />
                                </div>
                                <p>{locale.shop_containers.shop_drop_zone_helper_p}</p>
                              </div>
                            )}

                            {cover == null && (
                              <div className="drop-zone-content">
                                <div className="is-flex aic mar-bot-10">
                                  <img
                                    alt="icon"
                                    className="mar-right-20"
                                    src={require("../assets/images/image-file-icon.svg")}
                                  />
                                  <img
                                    alt="icon"
                                    src={require("../assets/images/drag-file-icon.svg")}
                                  />
                                </div>
                                <p>{locale.shop_containers.shop_drop_zone_helper_p}</p>
                              </div>
                            )}
                            <p />
                          </div>
                        </section>
                      )}
                    </Dropzone>
                    <CustomCloseBtn onClick={removeCover} isVisible={cover} />
                  </div>
                  <div className="columns is-flex is-vcentered is-mobile is-multiline">
                    <div className="column is-4-desktop is-8-mobile">
                      <div className="choose-cover">
                        <TButton
                          text={locale.choose_picture_placeholder}
                          className="blue outline"
                          id={"cover-button"}
                          name="cover"
                          onPress={() => this.refs.changeCover.click()}
                          ariaLabel="Upload Cover Photo"
                        />
                        <input
                          onChange={e => onDrop(e)}
                          type="file"
                          ref="changeCover"
                          className="is-hidden"
                        />
                      </div>
                    </div>
                    <div className="column is-6-desktop is-12-mobile">
                      <p>{locale.shop_containers.shop_cover_helper_p}</p>
                    </div>
                  </div>
                  {startShop && (
                    <label className="is-flex aic mar-bot-20">
                      <Checkbox
                        ischecked={agreeTerms}
                        onClick={() => {
                          this.setState({
                            agreeTerms: !agreeTerms
                          });
                        }}
                      />
                      <span className="checkbox-text">
                        {locale.shop_containers.shop_checkbox_agreement_text}{" "}
                        <a className="blue" href="#">
                          {locale.shop_checkbox_terms}
                        </a>
                      </span>
                    </label>
                  )}

                  <div className="columns is-mobile">
                    <div className="column is-4-desktop" />
                    <div className="column is-4-desktop">
                      <TButton
                        text={locale.cancel}
                        className="red outline full"
                        onPress={() => this.props.history.goBack()}
                        ariaLabel="Cancel"
                      />
                    </div>
                    <div className="column is-4-desktop">
                      <TButton
                        text={
                          startShop
                            ? locale.shop_containers.start_shop_addresses_table_start
                            : locale.shop_addresses_table_save
                        }
                        // className={`grad-blue full button ${!mobile_verified ? "tooltip" : ""}`}
                        className={`grad-blue full button`}
                        // tip={locale.plz_verify}
                        loading={is_busy}
                        onPress={startShop ? startYourShop : editYourShop}
                        id={"shopForm-button"}
                        name="shopForm"
                        disabled={!this.state.agreeTerms}
                        ariaLabel="submit"
                      />
                    </div>
                  </div>

                  {/* <div className="mar-bot-15">
                    <ReCAPTCHA
                      sitekey={google_capcha}
                      onChange={this.verify_robot}
                    />
                  </div> */}
                </div>
              </div>
            </section>
            {editShop && (
              <AddressTable
                locale={locale}
                withEditLink
                address={address}
                delAddress={delAddress}
                addNewAddress={updateAddress}
                lang={this.props.lang}
                lang_id={this.props.lang_id}
              />
            )}
          </div>
        </div>
      </div>
    );
  }
}

export default ShopForm;
