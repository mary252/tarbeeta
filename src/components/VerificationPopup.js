import React, { Component } from "react";
import { Modal } from "../components/Layout";
import { TextInput, TButton } from "./Form";
import {
  sendVerificationEmail,
  sendMobileConfirmationCode
} from "../services/auth";
import { verifyEmail, verifyMobile } from "../services";

class VerificationPopup extends Component {
  state = {
    is_busy: false,
    email_verification_code: null,
    is_sending: false,
    error_msg: null
  };

  toggle = () => {
    this.modal.toggle();
    if (!this.modal.state.visible) {
      this.send_message();
    }
  };

  store_data = (event, input) => {
    this.setState({
      [input]: event.target.value
    });
  };

  verify = async () => {
    this.setState({
      is_busy: true
    });
    const { email_verification_code } = this.state;
    const { isVerifyMobile } = this.props;

    let func = !isVerifyMobile
      ? verifyMobile({ code: email_verification_code })
      : verifyEmail({ code: email_verification_code });
    await func
      .then(response => {
        if (response.status) {
          alert(
            this.props.translation.account_settings_updated_message
              ? this.props.translation.account_settings_updated_message
                  .success_message
              : ""
          );
          window.location.reload();
        }
      })
      .catch(err => {
        this.setState({
          error_msg: err.data ? err.data.message : "",
          is_busy: false
        });
      });
  };

  send_message = async () => {
    this.setState({
      is_sending: true
    });
    try {
      let response =
        this.props.code == 1
          ? await sendMobileConfirmationCode()
          : await sendVerificationEmail();
      if (response.status) {
        this.setState({
          is_sending: false
        });
      }
    } catch (err) {
      this.setState({
        error_msg:
          undefined !== err.response.data.error
            ? err.response.data.error
            : err.response.data.message,
        is_sending: false
      });
    }
  };

  render() {
    const { is_busy } = this.state;
    const { isVerifyMobile, translation } = this.props;
    return (
      <Modal ref={m => (this.modal = m)}>
        <div onClick={e => e.stopPropagation()} className="address-popup">
          <div className="address-popup-div">
            <p className="address-popup-header">
              {translation.verify_mobile.title}
            </p>
            <div className="columns is-mobile">
              <TextInput
                type="number"
                onKeyUp={e => this.store_data(e, "email_verification_code")}
                onKeyPress={e => {
                  if (e.key === "Enter") {
                    this.verify();
                  }
                }}
                placeholder={
                  translation.email_verification.popup_input_placeholder
                }
              />
            </div>

            <div className="columns is-mobile">
              <div className="column is-4">
                <TButton
                  text={translation.cancel}
                  onPress={this.toggle}
                  className="red outline full"
                  ariaLabel="Cancel"
                />
              </div>
              <div className="column is-2 " />
              <div className="column is-6">
                <TButton
                  text={translation.email_verification.verify_button}
                  onPress={() => this.verify()}
                  className="grad-blue full button"
                  loading={is_busy}
                  ariaLabel="Verify"
                />
              </div>
            </div>
          </div>
        </div>
      </Modal>
    );
  }
}

export default VerificationPopup;
