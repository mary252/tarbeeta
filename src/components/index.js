import Accordin from "./Accordin";
import Autocomplete from "./Autocomplete";
import BreadCrumb from "./BreadCrumb";
import EmptyPage from "./EmptyPage/EmptyPage";
import Faq from "./Faq";
import FiltersMobile from "./FiltersMobile/FiltersMobile";
import FollowersPopup from "./FollowersPopup/FollowersPopup";
import Footer from "./Footer/Footer";
import Header from "./Header/Header";

import OrdersItems from "./Orders/OrdersItems";
import OrdersItemsInfo from "./Orders/OrdersItemsInfo";
import StarsRating from "./Orders/StarsRating";

import Filters from "./Products/Filters";
import PlateColors from "./Products/PlateColors";
import ProductCard from "./Products/ProductCard";
import ProductDetailsMobile from "./Products/ProductDetailsMobile";
import ProductDetailsSection from "./Products/ProductDetailsSection";
import Slider from "./Products/Slider";

import Cover from "./Shop/Cover";
import ShopCard from "./Shop/ShopCard";

import AddressPopUp from "./AddressPopUp";
import AddressTable from "./AddressTable";
import AddToCartPopUp from "./AddToCartPopUp";
import BankPopUp from "./BankPopUp";
import BuyNow from "./BuyNow";
import CancelOrderPopUp from "./CancelOrderPopUp";
import ConfirmationPopup from "./ConfirmationPopup";
import CustomNotification from "./CustomNotification";
import Empty from "./Empty";
import InfiniteScroll from "./InfiniteScroll";
import NotFound from "./NotFound";
import NotificationList from "./NotificationList";
import NotificationRow from "./NotificationRow";
import Page from "./Page";
import ProductDetailsPopUp from "./ProductDetailsPopUp";
import ShopForm from "./ShopForm";
import Slides from "./Slides";
import VerificationPopup from "./VerificationPopup";
import WriteReviewPopUp from "./WriteReviewPopUp";
import Loader from "./Loader";
import LoaderDots from "./Loader/LoaderDots";
import LoaderEllipsis from "./Loader/LoaderEllipsis";
import Mainloader from "./Loader/Mainloader";
export * from "./Form";
export * from "./HOC";
export * from "./Cart";
export * from "./Checkout";
export * from "./Layout";
export * from "./SVG";
export * from "./PlaceHolder";

export {
  Accordin,
  Autocomplete,
  BreadCrumb,
  EmptyPage,
  Faq,
  FiltersMobile,
  FollowersPopup,
  Footer,
  Header,
  OrdersItems,
  OrdersItemsInfo,
  StarsRating,
  Filters,
  PlateColors,
  ProductCard,
  ProductDetailsMobile,
  ProductDetailsSection,
  Slider,
  Cover,
  ShopCard,
  AddressPopUp,
  AddressTable,
  AddToCartPopUp,
  BankPopUp,
  BuyNow,
  CancelOrderPopUp,
  ConfirmationPopup,
  CustomNotification,
  Empty,
  InfiniteScroll,
  NotFound,
  NotificationList,
  NotificationRow,
  Page,
  ProductDetailsPopUp,
  ShopForm,
  Slides,
  VerificationPopup,
  WriteReviewPopUp,
  Loader,
  LoaderDots,
  LoaderEllipsis,
  Mainloader
};
