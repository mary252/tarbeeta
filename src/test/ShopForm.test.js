import React from "react";
import { shallow } from "enzyme";
import { ShopForm } from "../components/ShopForm";
import { StartShop } from "../containers/Shop/StartShop";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { baseURL } from "../common/Config";
import { postNewShop } from "../services";
import axios from "axios"; // v0.15.3
import httpAdapter from "axios/lib/adapters/http";
import nock from "nock";
import VerificationPopup from "../components/VerificationPopup";
configure({ adapter: new Adapter() });
jest.setTimeout(30000);


const match = {
  params: {
    lang: "en" //any id you want to set
  }
};
const location = {
  search: "",
  pathname: "/en/start"
};

const host = baseURL;

axios.defaults.host = host;
axios.defaults.adapter = httpAdapter;
let query = {
  lang_id: 1
};
global.URL.createObjectURL = jest.fn();

const props ={
    changeAvatar:jest.fn()
}

const startShop = shallow(
    <StartShop 
        match={match}
        location={location}
    />
);

let startShopInstance = startShop.instance();



const formShop = shallow(
  <ShopForm
    match={match}
    mobile_verified={1}
    mobile={"0102357890"}
    location={location}
    onDrop={startShopInstance.onDrop}
    onChange={startShopInstance.onChange}
    locale={startShopInstance.state.locale}
    {...props}
  />
);
let instance = formShop.instance();

describe('Shop Form container', () => {
    it('shop form renders', () => {
        expect(formShop.exists()).toBe(true);
    });

    it('renders correctly', () => {
        expect(ShopForm).toMatchSnapshot();
    });

    it('the locale load', () => {
        expect(instance.state.locale).not.toBeNull();
    });

    // it('verify robot function test ',()=>{
    //     expect(instance.state. verify_robot).toBeFalsy()
    //     instance.verify_robot('12321')
    //     expect(instance.state. verify_robot).toBeTruthy()

    // })

    it('toogle busy test ',()=>{
        expect(instance.state.is_busy).toBeFalsy()
        instance.toggle_busy()
        expect(instance.state.is_busy).toBeTruthy()
    })

    it('change avatar function test',()=>{
        expect(startShopInstance.state.avatar).toBe("")
        let e={
            target:{
                files:['file']
            }
        }
        startShopInstance.changeAvatar(e)
        expect(startShopInstance.state.avatar).toBe("file")
    })

    it('ondrop function test',()=>{
        expect(startShopInstance.state.cover).toBe('')
        let e={
            target:{
                files:['file']
            }
        }
        startShopInstance.onDrop(e)
        expect(startShopInstance.state.cover).toBe('file')
    })

    it('onchange function test',()=>{
        expect(startShopInstance.state.name).toBe('')
        let e={
            target:{
                name:'name',
                value:'name'
            }
        }
        startShopInstance.onChange(e)
        expect(startShopInstance.state.name).toBe('name')
    })
});


describe("shop form functional testing", async () => {
    it('name validate',async()=>{
        startShopInstance.setState({
            name:'name'
        })
       let res=await startShopInstance.validation(['name'])
       expect(res).toBe(true)
       
    })

    const startShop_copy = shallow(<StartShop match={match} location={location}/>);
        let another_instance=startShop_copy.instance()
        another_instance.setState({
            name:'name',
            verify_robot:true
        })
    await another_instance.startYourShop()
    expect(another_instance.state.redirect).toBeTruthy()
    let dis=localStorage.getItem("destination")
    expect(dis).toEqual('start')

});


describe("buttons test", () => {
    it("change avatar test", () => {
        instance.setState({loggedin: true, redirect: false});
    //   startShopInstance.setState({ loggedin: true });
      const avatar_button = formShop.find("#avatar-button");
      expect(avatar_button.length).toEqual(1);
    // expect(instance.props.changeAvatar).toHaveBeenCalled();

    });
  
    // it('name input test',()=>{
    //     const username=Shop.find("#name-input")
    //     expect(username.length).toEqual(1);
    //     const event = { target: { name: "name", value: "456" } };
    //     username.simulate("keyUp", event);
    //     expect(instance.state.email).toEqual("456");
  
    //   })
  });
  