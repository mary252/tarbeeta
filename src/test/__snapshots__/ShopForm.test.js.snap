// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop Form container renders correctly 1`] = `[Function]`;
